weather.js

```diff
@@ -5,17 +5,31 @@ dotenv.config();
 const apiURLPrefix = "http://api.openweathermap.org/";
 
 /**
- * Runs a conditional to distinguish if Zipcode or address string was entered.
+ * Get weather data based on whether Zipcode or address string was entered.
  * @param {string|number}input
  * @returns {Promise<void>}
  */
 async function getWeather(input) {
-    if (isNaN(input)) {
-        const data = await getCoordinates(input);
-        await getWeatherData(`lat=${data[0]["lat"]}&lon=${data[0]["lon"]}`);
-        return;
+    try {
+        formatCheck(input);
+        if (isNaN(input)) {
+            const data = await getCoordinates(input);
+            return printWeather(await getWeatherData(`lat=${data[0]["lat"]}&lon=${data[0]["lon"]}`));
+        }
+        printWeather(await getWeatherData(`zip=${input}`));
+    } catch (error) {
+        console.log(customErrorHandler(error));
+    }
+}
+
+/**
+ * Checks user input format
+ * @param {string|number}input
+ */
+function formatCheck(input) {
+    if (input.indexOf(',') >= 2) {
+        throw 'format error';
```

Comment:
We want a custom exception - look at this - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw#throw_an_object